testing/citest/tests/kube_smoke_test.py

```diff
@@ -389,6 +389,324 @@ def make_deploy_stage(self, imageSource=None, requisiteStages=[], **kwargs):
+          limitConcurrent='true',
+          parallel='true',
+          stages=[deploy_stage]
+      )
+      payload = self.agent.make_json_payload_from_kwargs(**job)
+      expect_match = {key: jp.EQUIVALENT(value)
+                      for key, value in job.items()}
+      expect_match['stages'] = jp.LIST_MATCHES(
+          [
+              jp.DICT_MATCHES({key: jp.EQUIVALENT(value)
+                               for key, value in deploy_stage.items()})])
+
+      builder = st.HttpContractBuilder(self.agent)
+      (builder.new_clause_builder('Has Pipeline',
+                                  retryable_for_secs=15)
+       .get_url_path(
+          'applications/{app}/pipelineConfigs'.format(app=self.TEST_APP))
+       .contains_match(expect_match))
+      return st.OperationContract(self.new_post_operation(title='create_delete daemonset', data=payload, path='pipelines',
+                                                          status_class=st.SynchronousHttpOperationStatus),
+                                  contract=builder.build())
+
+  def make_delete_stage_daemonset(self, **kwargs):
+      bindings = self.bindings
+      result = {
+          'refId': '1',
+          'type': 'destroyServerGroup',
+          'cloudProvider': 'kubernetes',
+		  'cloudProviderType': 'kubernetes',
+		  'cluster': self.TEST_APP+'tstv001',
+		  'credentials': bindings['SPINNAKER_KUBERNETES_ACCOUNT'],
+          'interestingHealthProviderNames': ['KubernetesService'],
+		  'kind': 'DaemonSet',
+		  'name': 'Destroy Server Group',
+                  'target': 'current_asg_dynamic',
+		  'namespaces': ['default'],
+		  'preCondition': 'true'
+      }
+      result.update(kwargs)
+      return result
+
+  def make_delete_stage_statefulset(self, **kwargs):
+      bindings = self.bindings
+      result = {
+          'refId': '1',
+          'type': 'destroyServerGroup',
+          'cloudProvider': 'kubernetes',
+		  'cloudProviderType': 'kubernetes',
+		  'cluster': self.TEST_APP+'tstv001',
+		  'credentials': bindings['SPINNAKER_KUBERNETES_ACCOUNT'],
+          'interestingHealthProviderNames': ['KubernetesService'],
+		  'kind': 'StatefulSet',
+                  'target': 'current_asg_dynamic',
+		  'name': 'Destroy Server Group',
+		  'namespaces': ['default'],
+		  'preCondition': 'true'
+      }
+      result.update(kwargs)
+      return result
+
+  def execute_delete_daemonset_pipeline(self):
+      bindings = self.bindings
+      payload = self.agent.make_json_payload_from_kwargs(
+          job=[{
+              'type': 'manual',
+              'user': '[anonymous]'
+          }],
+          description='Delete Daemonset in Application' + self.TEST_APP,
+          application=self.TEST_APP)
+      builder = kube.KubeContractBuilder(self.kube_observer)
+      (builder.new_clause_builder('Daemonset Deleted', retryable_for_secs=15)
+       .get_resources(
+          'daemonsets', extra_args=[self.TEST_APP+'tstv001', '--namespace', self.TEST_NAMESPACE])
```

Comment:
I dont understand what this is testing. I guess I dont understand the feature either which doesnt help. With a delete operation, I'd expect to test for the absense of something, or an error trying to look it up. I wouldnt expect to see a value ("kind=DaemonSet") similar to what I just deleted. If this is in fact desired, please explain what this is testing with a comment.

Reply:
Yes you are right , i have changed the code as per already did in 'delete_server_group'.